flutter/lib/ui/geometry.dart

```diff
@@ -680,35 +680,183 @@ class Rect {
+class Radius {
+  /// Constructs a circular radius.
+  const Radius.circular(double radius) : this.oval(radius, radius);
+
+  /// Construct an oval radius.
+  const Radius.oval(this.x, this.y);
+
+  /// A radius with the value zero.
+  static final Radius zero = new Radius.circular(0.0);
+
+  /// The radius value on the horizontal x axis.
+  final double x;
+
+  /// The radius value on the vertical y axis.
+  final double y;
+
+  /// Linearly interpolate between two radii.
+  ///
+  /// If either is null, this function substitutes [Radius.zero] instead.
+  static Radius lerp(Radius a, Radius b, double t) {
+    if (a == null && b == null)
+      return null;
+    if (a == null)
+      return new Radius.oval(b.x * t, b.y * t);
+    if (b == null) {
+      double k = 1.0 - t;
+      return new Radius.oval(a.x * k, a.y * k);
+    }
+    return new Radius.oval(
+      lerpDouble(a.x, b.x, t),
+      lerpDouble(a.y, b.y, t)
+    );
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! Radius)
+      return false;
+    final Radius typedOther = other;
+    return typedOther.x == x && typedOther.y == y;
+  }
+
+  @override
+  int get hashCode => hashValues(x, y);
+
+  @override
+  String toString() {
+    return x == y ? 'Radius.circular($x)' : 'Radius.oval($x, $y)';
+  }
+}
+
+/// A rounded rectangle with the custom radii for all four corners.
 class RRect {
   RRect._();
 
   /// Construct a rounded rectangle from its left, top, right, and bottom edges,
-  /// and the radii along its horizontal axis and its vertical axis.
-  RRect.fromLTRBXY(double left, double top, double right, double bottom, double radiusX, double radiusY) {
+  /// and the constant radii along its horizontal axis and its vertical axis.
+  RRect.fromLTRBXY(double left, double top, double right, double bottom,
+                   double radiusX, double radiusY) {
     _value
       ..[0] = left
       ..[1] = top
       ..[2] = right
       ..[3] = bottom
       ..[4] = radiusX
-      ..[5] = radiusY;
+      ..[5] = radiusY
+      ..[6] = radiusX
+      ..[7] = radiusY
+      ..[8] = radiusX
+      ..[9] = radiusY
+      ..[10] = radiusX
+      ..[11] = radiusY;
+  }
+
+  /// Construct a rounded rectangle from its left, top, right, and bottom edges,
+  /// and a radius that is constant in each corner.
+  RRect.fromLTRBR(double left, double top, double right, double bottom,
+                      Radius radius) {
+    _value
+      ..[0] = left
+      ..[1] = top
+      ..[2] = right
+      ..[3] = bottom
+      ..[4] = radius.x
+      ..[5] = radius.y
+      ..[6] = radius.x
+      ..[7] = radius.y
+      ..[8] = radius.x
+      ..[9] = radius.y
+      ..[10] = radius.x
+      ..[11] = radius.y;
   }
 
-  /// Construct a rounded rectangle from its bounding box and the radii along
-  /// its horizontal axis and its vertical axis.
+  /// Construct a rounded rectangle from its bounding box and the constant radii
```

Comment:
s/constant/same/ maybe?

constant means something different in this context than what we're trying to say here